apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/ExperimentsResourceTest.java

```diff
@@ -1345,6 +1346,70 @@ private Experiment createExperimentWithFeedbackScores(String apiKey, String work
                             .toList())
                     .build();
         }
+
+        @ParameterizedTest
+        @MethodSource
+        @DisplayName("when searching by prompt id and result having {} experiments, then return page")
+        void find__whenSearchingByPromptIdAndResultHavingXExperiments__thenReturnPage(int experimentCount,
+                int expectedMatchCount) {
+
+            var workspaceName = UUID.randomUUID().toString();
+            var workspaceId = UUID.randomUUID().toString();
+            var apiKey = UUID.randomUUID().toString();
+
+            mockTargetWorkspace(apiKey, workspaceName, workspaceId);
+
+            var dataset = podamFactory.manufacturePojo(Dataset.class);
+            createAndAssert(dataset, apiKey, workspaceName);
+
+            IntStream.range(0, experimentCount - expectedMatchCount)
+                    .parallel()
+                    .forEach(i -> {
+                        var experiment = generateExperiment().toBuilder()
+                                .datasetName(dataset.name())
+                                .build();
+
+                        createAndAssert(experiment, apiKey, workspaceName);
+                    });
+
+            String promptId = UUID.randomUUID().toString();
+
+            CreatePromptVersion request = CreatePromptVersion.builder()
+                    .name(promptId)
+                    .version(podamFactory.manufacturePojo(PromptVersion.class))
+                    .build();
+
+            PromptVersion promptVersion = createPromptVersion(request, apiKey, workspaceName);
```

Comment:
Can be moved to a function to avoid duplication, almost same in the other test.

Reply:
That is exactly the difference between the 2 tests, this one creates and prompts while the other doesn't